### DISCUSSION TITLE
Targeting the Semaphore libraries to `ES6`

### DISCUSSION THREAD

**Discussion by cedoor** [semaphore-protocol/semaphore#644] [General]
The current Semaphore libraries are written in TypeScript and are "compiled" into JavaScript with Rollup and `tsc`. The target in the current `tsconfig.ts` configuration is ES5, i.e. the first major revision to JavaScript.

Targeting ES6 (ECMAScript 2015) should be generally fine today. It's widely supported across modern browsers and environments and it would allow Semaphore bundles to use many ES6 features natively, like arrow functions, promises, and classes, which can improve code readability and efficiency.

**Comment by vplasencia** [comment]
Hey! I agree. We could also use a version later than `ECMAScript 2015`.

**Reply by cedoor** [reply]
Most modern browsers should also support `ES2020` features. I took a look at https://caniuse.com, and it seems to confirm it. Users who are currently using browsers that don't support `ES2020` are usually < 1%.
